test/mdset.c

```diff
@@ -0,0 +1,714 @@
+                                erbufi[k][m][n] = efbufi[k][m][n];
+                    } /* end if */
+                    else
+                        for (m = 0; m < dset_dims[k][0]; m++)
+                            for (n = 0; n < dset_dims[k][1]; n++)
+                                efbufi[k][m][n] = wbufi[k][m][n];
+                }
+            } /* end for */
+
+            /* Perform I/O */
+            if (do_read) {
+                if (flags & MDSET_FLAG_MDSET) {
+                    /* Set buffers */
+                    for (k = 0; k < ndsets; k++)
+                        rbufs[k] = rbufi[k][0];
+
+                    /* Read datasets */
+                    if (H5Dread_multi(ndsets, dset_ids, mem_type_ids, mem_space_ids, file_space_ids,
+                                      H5P_DEFAULT, rbufs) < 0)
+                        TEST_ERROR;
+                } /* end if */
+                else
+                    /* Read */
+                    if (H5Dread(dset_ids[0], mem_type_ids[0], mem_space_ids[0], file_space_ids[0],
+                                H5P_DEFAULT, rbuf) < 0)
+                        TEST_ERROR;
+
+                /* Verify data */
+                if (0 != memcmp(rbuf, erbuf, buf_size))
+                    TEST_ERROR;
+            } /* end if */
+            else {
+                if (flags & MDSET_FLAG_MDSET) {
+                    /* Set buffers */
+                    for (k = 0; k < ndsets; k++)
+                        wbufs[k] = wbufi[k][0];
+
+                    /* Write datasets */
+                    if (H5Dwrite_multi(ndsets, dset_ids, mem_type_ids, mem_space_ids, file_space_ids,
+                                       H5P_DEFAULT, wbufs) < 0)
+                        TEST_ERROR;
+                } /* end if */
+                else
+                    /* Write */
+                    if (H5Dwrite(dset_ids[0], mem_type_ids[0], mem_space_ids[0], file_space_ids[0],
+                                 H5P_DEFAULT, wbuf) < 0)
+                        TEST_ERROR;
+
+                /* Update wbuf */
+                for (l = 0; l < max_dsets; l++)
+                    for (m = 0; m < MAX_DSET_X; m++)
+                        for (n = 0; n < MAX_DSET_Y; n++)
+                            wbufi[l][m][n] += (unsigned)max_dsets * MAX_DSET_X * MAX_DSET_Y;
+            } /* end else */
+        }     /* end for */
+
+        /* Close */
+        for (j = 0; j < ndsets; j++) {
+            if (H5Dclose(dset_ids[j]) < 0)
+                TEST_ERROR;
+            dset_ids[j] = -1;
+            if (H5Sclose(file_space_ids[j]) < 0)
+                TEST_ERROR;
+            file_space_ids[j] = -1;
+        } /* end for */
+        if (H5Fclose(file_id) < 0)
+            TEST_ERROR;
+        file_id = -1;
+
+        /* Cleanup external file.  Need to do this because otherwise there is garbage when the
+         * dataset is created, even with early allocation and fill time. */
+        HDremove(EXT_FILENAME);
```

Comment:
Might consider using H5Fdelete instead of HDremove for compatibility with different file drivers

Reply:
According to the doxygen, H5Fdelete only works with HDF5 files. As the external file is just a data blob, it won't pass the H5Fis_accessible() check.

Reply:
Ah ok. I guess that may leave external files as being a bit problematic for H5Fdelete